File: examples/test-projects/small-projects/test-project-python/string_processor.py
"""Обработчик строк — аналог StringProcessor из других малых проектов."""


class StringProcessor:
    """Класс для базовых операций над строками."""

    def __init__(self):
        self.processed_count = 0

    def to_upper(self, s):
        result = s.upper()
        self._increment_processed()
        return result

    def to_lower(self, s):
        result = s.lower()
        self._increment_processed()
        return result

    def reverse(self, s):
        result = s[::-1]
        self._increment_processed()
        return result

    def is_palindrome(self, s):
        reversed_s = self.reverse(s)
        return s == reversed_s

    def contains(self, s, ch):
        self._increment_processed()
        return ch in s

    def count_chars(self, s, ch):
        count = 0
        for c in s:
            if c == ch:
                count += 1
        self._increment_processed()
        return count

    def word_count(self, s):
        if not s:
            return 0
        count = 1
        for ch in s:
            if ch == ' ':
                count += 1
        return count

    def _increment_processed(self):
        self.processed_count += 1

File: examples/test-projects/small-projects/test-project-java/StringProcessor.java
package testproject;

/** Обработчик строк — аналог C++ StringProcessor. */
public class StringProcessor {
    private int processedCount = 0;

    public String toUpper(String str) {
        String result = str.toUpperCase();
        incrementProcessed();
        return result;
    }

    public String toLower(String str) {
        String result = str.toLowerCase();
        incrementProcessed();
        return result;
    }

    public String reverse(String str) {
        String result = new StringBuilder(str).reverse().toString();
        incrementProcessed();
        return result;
    }

    public boolean isPalindrome(String str) {
        String reversed = reverse(str);
        return str.equals(reversed);
    }

    public boolean contains(String str, char c) {
        incrementProcessed();
        return str.indexOf(c) >= 0;
    }

    public int countChars(String str, char c) {
        int count = 0;
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == c) {
                count++;
            }
        }
        incrementProcessed();
        return count;
    }

    public int wordCount(String str) {
        if (str.isEmpty()) {
            return 0;
        }
        int count = 1;
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == ' ') {
                count++;
            }
        }
        return count;
    }

    public void incrementProcessed() {
        processedCount++;
    }
}
